src/Models/NonhydrostaticModels/nonhydrostatic_model.jl

```diff
@@ -146,18 +146,7 @@ function NonhydrostaticModel(; grid,
 
     if hydrostatic_pressure_anomaly isa DefaultHydrostaticPressureAnomaly
         # Manage treatment of the hydrostatic pressure anomaly:
-        
-        if grid isa ImmersedBoundaryGrid
-            # Separate the hydrostatic pressure anomaly
-            # from the nonhydrostatic pressure contribution.
-            # See https://github.com/CliMA/Oceananigans.jl/issues/3677.
-            
-            hydrostatic_pressure_anomaly = CenterField(grid)
-        else
-            # Use a single combined pressure, saving memory and computation.
-            
-            hydrostatic_pressure_anomaly = nothing
-        end
```

Comment:
I suggest restorring this code and writing 

```julia
if !isnothing(buoyancy)
```

instead of 

```julia
if grid isa ImmersedBoundaryGrid
```

remember there is  still no motivation to separate out the hydrostatic pressure if `b=0`.

Reply:
(even on `ImmersedBoundaryGrid` of course)